src/Uno.UI/UI/Xaml/Controls/ItemsControl/ItemsControl.cs

```diff
@@ -1041,9 +1039,16 @@ internal DependencyObject GetContainerForIndex(int index)
 		{
 			var item = ItemFromIndex(index);
 
-			var container = IsItemItsOwnContainerOverride(item)
-				? item as DependencyObject
-				: GetContainerForItemOverride();
+			var container = default(DependencyObject);
+
+			if (IsItemItsOwnContainerOverride(item))
+			{
+				container = item as DependencyObject;
+			}
+			else
+			{
+				container = ItemsPanelRoot?.Children?.OfType<ContentPresenter>()?.FirstOrDefault(c => c.Content?.Equals(item) ?? false) ?? GetContainerForItemOverride();
```

Comment:
```suggestion
				container = ItemsPanelRoot?.Children
				  ?.OfType<ContentPresenter>()
				  ?.FirstOrDefault(c => object.ReferenceEquals(c.Content, item) ?? false) 
				  ?? GetContainerForItemOverride();
```

Reply:
We need to compare references, not using the Equals method (for which an Type may have the `Equals` overridden)

Reply:
So... an ItemsControl will recycle containers now? That's a pretty sharp departure from UWP. And normally the template shouldn't be recreated, that suggests a bug regarding the FrameworkTemplate pooling.

Also to add to Jerome's comment, what's going to happen if the same item (by ref) occurs twice in the ItemsSource?

Reply:
I will cancel that PR until we get a clearer view  of what should be done. Anyway, there is a perf issue to fix here ;)